bctl/agent/main.go

```diff
@@ -95,19 +98,33 @@ func main() {
 	}
 
 	if err != nil {
+		// The agent actually starts in agent.Run(), this message simplifies it to the user
 		fmt.Printf("ERROR: failed to start agent: %s\n", err)
 		os.Exit(1)
 	}
 
-	if exitError := agent.Run(); exitError == nil {
-		os.Exit(0)
+	if agentService := NewAgentService(agent); agentService == nil {
+		os.Exit(1)
+	} else if exitError := agentService.Run(); exitError != nil {
+		os.Exit(1)
 	}
-
-	os.Exit(1)
+	os.Exit(0)
```

Comment:
IMO there's no need for this check; we've already verified that `agent` is non-nil. 

```suggestion
        agentService := NewAgentService(agent)
	if exitError := agentService.Run(); exitError != nil {
		os.Exit(1)
	}
	os.Exit(0)
```

Reply:
Not sure I follow, this checks whether any error has came up from the agent service init. How is checking the agent relevant?

Reply:
Ah totally right
a25d06d